internal/gatewayapi/route.go

```diff
@@ -846,24 +846,22 @@ func (t *Translator) processUDPRouteParentRefs(udpRoute *UDPRouteContext, resour
 			)
 			continue
 		}
-		if len(udpRoute.Spec.Rules[0].BackendRefs) != 1 {
-			parentRef.SetCondition(udpRoute,
-				gwapiv1.RouteConditionResolvedRefs,
-				metav1.ConditionFalse,
-				"InvalidBackend",
-				"One and only one backend is supported",
-			)
-			continue
-		}
 
-		backendRef := udpRoute.Spec.Rules[0].BackendRefs[0]
-		ds, _ := t.processDestination(backendRef, parentRef, udpRoute, resources)
-		// Skip further processing if route destination is not valid
-		if ds == nil || len(ds.Endpoints) == 0 {
+		valid := true
+		for _, backendRef := range udpRoute.Spec.Rules[0].BackendRefs {
+			ds, _ := t.processDestination(backendRef, parentRef, udpRoute, resources)
+			// Skip further processing if route destination is not valid
+			if ds == nil || len(ds.Endpoints) == 0 {
+				valid = false
+				continue
+			}
+
+			destSettings = append(destSettings, ds)
+		}
+		if !valid {
 			continue
```

Comment:
we should add a status for the `invalid` case

Reply:
correct me if I'm wrong, looks like we didn't do that thing for all routes?